packages/@aws-cdk/aws-ecr/lib/lifecycle.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * An ECR life cycle rule
+ */
+export interface LifecycleRule {
+    /**
+     * Controls the order in which rules are evaluated (low to high)
+     *
+     *  When you add rules to a lifecycle policy, you must give them each a
+     *  unique value for rulePriority. Values do not need to be sequential
+     *  across rules in a policy. A rule with a tagStatus value of any must have
+     *  the highest value for rulePriority and be evaluated last.
+     */
+    rulePriority: number;
```

Comment:
I guess the common case would be to define a lifecycle rule for all images (tagStatus == any), in which case rulePriority must be the maximum value out of all rules). Perhaps if `rulePriority` is `undefined`, it will automatically get a priority allocated for it (max(rules)+1) or something. This way, we can make this optional and the common case more ergonomic.

Reply:
Yeah, I wasn't really sure of how to do this in an elegant way yesterday, but I think we can make both work.